File: Unit2-01/BasicClass.java
/*
 * This class creates an array of integers
 * @author Kaitlyn Ip
 * @version 1.0
 * @since   2024-03-08
 */

/**
 * This is the class.
 */
public class BasicClass {
    /**
     * Adds strings to an array.
     *
     * @param array contains the array of strings
     */
    private String[] array = {};

    /**
     * Pushes the string to an array.
     *
     * @param string the string that is going to be pushed
     */
    public void push(String string) {
        final String[] newArray = new String[this.array.length + 1];

        for (int counter = 0; counter < this.array.length; counter++) {
            newArray[counter] = this.array[counter];
        }

        newArray[this.array.length] = string;
        this.array = newArray;
    }

    /**
     * Prints the array to the console.
     *
     * @return the values of the array formatted
     */
    public String showStack() {
        String stackValues = this.array[0];

        for (int counter = 1; counter < this.array.length; counter++) {
            stackValues = stackValues + ", " + this.array[counter];
        }
        return stackValues;
    }
}

File: Unit2-01/BasicClass.ts
/*
 * This class creates a list of strings
 * @author Kaitlyn Ip
 * @version 1.0
 * @since 2024-03-07
 */

export class BasicClass {
    // Array that contains all of the string
    private array: string[]

    // Creates the array
    public constructor() {
      this.array = []
    }

    // Pushes the strings to the arary
    public push(string: string) {
      this.array.push(string)
    }

    // Prints the array to the console
    public showStack() {
      let stackValues = this.array[0]

      for (let counter = 1; counter < this.array.length; counter++) {
        stackValues = stackValues + ", " + this.array[counter]
      }

      return stackValues
  }
}
